src/frontend/Typechecker.ml

```diff
@@ -976,10 +976,11 @@ let verify_assignment_global loc cf block is_global id =
 (* Until function types are added to the user language, we
    disallow assignments to function values
 *)
-let verify_assignment_non_function loc ut id =
+let rec verify_assignment_non_function loc ut =
   match ut with
   | UnsizedType.UFun _ | UMathLibraryFunction ->
-      Semantic_error.cannot_assign_function loc ut id.name |> error
+      Semantic_error.cannot_assign_function loc ut |> error
+  | UTuple ts -> List.iter ~f:(verify_assignment_non_function loc) ts
   | _ -> ()
```

Comment:
This function does not need to change if it's called from the same place as `verify_assignable_id` (ie. on line 1173). I think the error is a lot nicer with the function name.

Reply:
that location (check_lvalue) does not have access to the rhs or its type, but I was able to move it to `check_assignment_operator` with the other typechecks